cpp/oneapi/dal/algo/louvain/common.hpp

```diff
@@ -0,0 +1,168 @@
+    void set_resolution(double value);
+    void set_max_iteration_count(std::int64_t value);
+
+    dal::detail::pimpl<descriptor_impl<Task>> impl_;
+};
+
+} // namespace detail
+
+/// Class for the Louvain algorithm descriptor
+///
+/// @tparam Float The data type of the result
+/// @tparam Method The algorithm method
+/// @tparam Task   The task to solve by the algorithm
+/// @tparam Allocator   Custom allocator for all memory management inside the
+/// algorithm
+template <typename Float = float,
+          typename Method = method::by_default,
+          typename Task = task::by_default,
+          typename Allocator = std::allocator<char>>
+class descriptor : public detail::descriptor_base<Task> {
+    static_assert(detail::is_valid_method<Method>);
+    static_assert(detail::is_valid_task<Task>);
+
+    using base_t = detail::descriptor_base<Task>;
+
+public:
+    using float_t = Float;
+    using method_t = Method;
+    using task_t = Task;
+    using allocator_t = Allocator;
+
+    descriptor(Allocator allocator = std::allocator<char>()) {
+        _alloc = allocator;
+    }
+
+    /// Returns minimum modularity increase inside iteration
+    double get_accuracy_threshold() const {
+        return base_t::get_accuracy_threshold();
+    }
+
+    /// Sets the minimum modularity increase indide iteration
+    ///
+    /// @param [in] accuracy_threshold  Minimum modularity increase indide
+    /// iteration
+    /// @invariant :expr:`accuracy_threshold >= 0`
+    /// @remark default = 0.0001
+    auto &set_accuracy_threshold(double accuracy_threshold) {
+        base_t::set_accuracy_threshold(accuracy_threshold);
+        return *this;
+    }
+
+    /// Returns resolution parameter in the modularity formula.
+    double get_resolution() const {
+        return base_t::get_resolution();
+    }
+
+    /// Sets resolution parameter in the modularity formula
+    ///
+    /// @param [in] resolution  Resolution parameter in the modularity formula
+    /// @invariant :expr:`resolution >= 0`
+    /// @remark default = 1
+    auto &set_resolution(double resolution) {
+        base_t::set_resolution(resolution);
+        return *this;
+    }
+
+    /// Returns the maximum number of iterations of the Louvain algorithm
+    std::int64_t get_max_iteration_count() {
+        return base_t::get_max_iteration_count();
+    }
+
+    /// Sets the maximum number of iterations of the Louvain algorithm
+    ///
+    /// @param [in] max_iteration_count  Maximum number of iterations of the
+    /// Louvain algorithm
+    /// @invariant :expr:`max_iteration_count >= 0`
+    /// @remark default = 0
+    auto &set_max_iteration_count(std::int64_t max_iteration_count) {
+        base_t::set_max_iteration_count(max_iteration_count);
+        return *this;
+    }
+
+    Allocator get_allocator() const {
+        return _alloc;
+    }
+
+private:
+    Allocator _alloc;
+};
+
+namespace detail {
+
+template <typename Graph>
+constexpr bool is_valid_graph =
```

Comment:
only undirected?

Reply:
it was decided that the initial version of the algorithm would work only with undirected graphs, because the modularity formula is slightly different for the oriented case